javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java

```diff
@@ -795,34 +793,97 @@ void apply(NodeText nodeText, Node node) {
         } while (diffIndex < this.elements.size() || nodeTextIndex < nodeText.getElements().size());
     }
 
-    private int findIndexOfCorrespondingNodeTextElement(CsmElement csmElement, NodeText nodeText, int startIndex, Set<Integer> usedIndexes, Node node) {
-        for (int i=startIndex;i<nodeText.getElements().size();i++){
-            if (!usedIndexes.contains(i)) {
-                TextElement textElement = nodeText.getTextElement(i);
-                if (csmElement instanceof CsmToken) {
-                    CsmToken csmToken = (CsmToken)csmElement;
-                    if (textElement instanceof TokenTextElement) {
-                        TokenTextElement tokenTextElement = (TokenTextElement)textElement;
-                        if (tokenTextElement.getTokenKind() == csmToken.getTokenType() && tokenTextElement.getText().equals(csmToken.getContent(node))) {
-                            usedIndexes.add(i);
-                            return i;
+    private List<Integer> findIndexOfCorrespondingNodeTextElement(List<CsmElement> elements, NodeText nodeText, int startIndex, Node node) {
```

Comment:
I really wonder how I ended up to write such complex code without comments and how someone managed to figure it out and improve it. Kudos!